plugins/configs/api/listener.go

```diff
@@ -0,0 +1,162 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log" // nolint:revive
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/models"
+)
+
+type ConfigAPIService struct {
+	server *http.Server
+	api    *api
+	Log    telegraf.Logger
+}
+
+func newConfigAPIService(server *http.Server, api *api, logger telegraf.Logger) *ConfigAPIService {
+	service := &ConfigAPIService{
+		server: server,
+		api:    api,
+		Log:    logger,
+	}
+	server.Handler = service.mux()
+	return service
+}
+
+// nolint:revive
+func (s *ConfigAPIService) mux() *mux.Router {
+	m := mux.NewRouter()
+	m.HandleFunc("/status", s.status).Methods("GET")
+	m.HandleFunc("/plugins/create", s.createPlugin).Methods("POST")
+	m.HandleFunc("/plugins/{id:[0-9a-f]+}/status", s.pluginStatus).Methods("GET")
+	m.HandleFunc("/plugins/list", s.listPlugins).Methods("GET")
+	m.HandleFunc("/plugins/running", s.runningPlugins).Methods("GET")
+	m.HandleFunc("/plugins/{id:[0-9a-f]+}", s.deleteOrUpdatePlugin).Methods("DELETE", "PUT")
+	return m
+}
+
+func (s *ConfigAPIService) status(w http.ResponseWriter, req *http.Request) {
+	if req.Body != nil {
+		defer req.Body.Close()
+	}
+	_, _ = w.Write([]byte("ok"))
+}
+
+func (s *ConfigAPIService) createPlugin(w http.ResponseWriter, req *http.Request) {
+	if req.Body != nil {
+		defer req.Body.Close()
+	}
+	cfg := PluginConfigCreate{}
+
+	dec := json.NewDecoder(req.Body)
+	if err := dec.Decode(&cfg); err != nil {
+		s.Log.Error("decode error %v", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	id, err := s.api.CreatePlugin(cfg)
+	if err != nil {
+		s.Log.Error("error creating plugin %v", err)
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
both are wrong. We need some better error handling to know which is correct. I'll make a note to come back to this later.

Reply:
improved client side error handling with this [commit](https://github.com/influxdata/telegraf/pull/9546/commits/1cdd3d8b3853953e5612e9943c7e2362e2314ce8) so now it wraps errors and if its a bad request or not found it will return that otherwise internal server error